File: ClrExtensions Auotmated Tests/EventsTest.vb
Imports System

Imports Microsoft.VisualStudio.TestTools.UnitTesting

Imports ClrExtensions



'''<summary>
'''This is a test class for EventsTest and is intended
'''to contain all EventsTest Unit Tests
'''</summary>
<TestClass()> _
Public Class EventsTest


    Private testContextInstance As TestContext

    '''<summary>
    '''Gets or sets the test context which provides
    '''information about and functionality for the current test run.
    '''</summary>
    Public Property TestContext() As TestContext
        Get
            Return testContextInstance
        End Get
        Set(ByVal value As TestContext)
            testContextInstance = Value
        End Set
    End Property

#Region "Additional test attributes"
    '
    'You can use the following additional attributes as you write your tests:
    '
    'Use ClassInitialize to run code before running the first test in the class
    '<ClassInitialize()>  _
    'Public Shared Sub MyClassInitialize(ByVal testContext As TestContext)
    'End Sub
    '
    'Use ClassCleanup to run code after all tests in a class have run
    '<ClassCleanup()>  _
    'Public Shared Sub MyClassCleanup()
    'End Sub
    '
    'Use TestInitialize to run code before running each test
    '<TestInitialize()>  _
    'Public Sub MyTestInitialize()
    'End Sub
    '
    'Use TestCleanup to run code after each test has run
    '<TestCleanup()>  _
    'Public Sub MyTestCleanup()
    'End Sub
    '
#End Region


    '''<summary>
    '''A test for Raise
    '''</summary>
    <TestMethod()> _
    Public Sub RaiseTest()
        Dim [event] As EventHandler = Nothing ' TODO: Initialize to an appropriate value
        Dim sender As Object = Nothing ' TODO: Initialize to an appropriate value
        Dim e As EventArgs = Nothing ' TODO: Initialize to an appropriate value
        Events.Raise([event], sender, e)
        Assert.Inconclusive("A method that does not return a value cannot be verified.")
    End Sub
End Class

File: MAIN/ClrExtensions CSharp Tests/EventsTest.cs
using ClrExtensions;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;

namespace Clr_Extensions_CSharp_Tests
{


	/// <summary>
	///This is a test class for EventsTest and is intended
	///to contain all EventsTest Unit Tests
	///</summary>
	[TestClass()]
	public class EventsTest
	{


		private TestContext testContextInstance;

		/// <summary>
		///Gets or sets the test context which provides
		///information about and functionality for the current test run.
		///</summary>
		public TestContext TestContext
		{
			get
			{
				return testContextInstance;
			}
			set
			{
				testContextInstance = value;
			}
		}

		#region Additional test attributes
		// 
		//You can use the following additional attributes as you write your tests:
		//
		//Use ClassInitialize to run code before running the first test in the class
		//[ClassInitialize()]
		//public static void MyClassInitialize(TestContext testContext)
		//{
		//}
		//
		//Use ClassCleanup to run code after all tests in a class have run
		//[ClassCleanup()]
		//public static void MyClassCleanup()
		//{
		//}
		//
		//Use TestInitialize to run code before running each test
		//[TestInitialize()]
		//public void MyTestInitialize()
		//{
		//}
		//
		//Use TestCleanup to run code after each test has run
		//[TestCleanup()]
		//public void MyTestCleanup()
		//{
		//}
		//
		#endregion




		[TestMethod()]
		public void RaiseTest()
		{
				var a = new EventTester();
				a.RaiseA();
				a.RaiseB();
				a.RaiseC();
				a.RaiseD();
				a.RaiseE();
				Assert.AreEqual(0, a.eventCount);

				var b = new EventTester();
				b.Event1 += b.Target1;
				b.Event2 += b.Target2;
				b.Event3 += b.Target4;

				b.RaiseA();
				Assert.AreEqual(1, b.eventCount);
				b.eventCount = 0;

				b.RaiseB();
				Assert.AreEqual(1, b.eventCount);
				b.eventCount = 0;

				b.RaiseC();
				Assert.AreEqual(2, b.eventCount);
				b.eventCount = 0;

				b.RaiseE();
				Assert.AreEqual(4, b.eventCount);
				b.eventCount = 0;

				b.RaiseE();
				Assert.AreEqual(4, b.eventCount);
				b.eventCount = 0;
		}


		private class EventTester
		{
			public int eventCount = 0;

			public void RaiseA()
			{
				Event1.Raise(this);
			}
			public void RaiseB()
			{
				Event1.Raise(this, EventArgs.Empty);
			}
			public void RaiseC()
			{
				Event2.Raise(this, new ResolveEventArgs ("XXX"));
			}
			public void RaiseD()
			{
				Event3.Raise(this);
			}
			public void RaiseE()
			{
				Event3.Raise(this, EventArgs.Empty);
			}

			public event EventHandler Event1;
			public event EventHandler<ResolveEventArgs> Event2;
			public event EventHandler<EventArgs> Event3;

			public void Target1(object source, EventArgs e)
			{
				eventCount += 1;
			}

			public void Target2(object source, ResolveEventArgs e)
			{
				eventCount += 2;
			}

			public void Target4(object source, EventArgs e)
			{
				eventCount += 4;
			}

			public void Target8(object source, ResolveEventArgs e)
			{
				eventCount += 8;
			}
		}


	}
}
